core/src/main/java/org/apache/calcite/sql/SqlCallBinding.java

```diff
@@ -411,4 +411,8 @@ public CalciteException newValidationError(
   public boolean isTypeCoercionEnabled() {
     return validator.config().typeCoercionEnabled();
   }
+
+  private SqlCall createDefaultCall() {
+    return SqlStdOperatorTable.DEFAULT.createCall(SqlParserPos.ZERO);
```

Comment:
Do we need to create a new call, if one already exists?

Reply:
having a single static call caused a circular dependency, the purpose is to avoid it (by creating the default call ad-hoc only when needed)

Reply:
Just in case, a nested static class would avoid circular dependency as well.

Reply:
Yes... but my fear is that it would be a more "obscure" solution when looking at this code in the future: "why is this static nested class here, for apparently no reason, just with a private static field...".
I guess it can be a good (and efficient) solution, but we'll need a clear javadoc stating the purpose of the class and why it is needed (and cannot be removed). It should be feasible, I'll commit this approach.